playbook/run_playbook/main.yml

```diff
@@ -0,0 +1,113 @@
+---
+- name: Start instance for playbook to run on
+  hosts: localhost
+  connection: local
+  vars:
+    ansible_ssh_pipelining: no
+    env: "dev"
+    azone: "az1"
+    sub_zone: "app"
+    sg_zone: "appserver"
+    env_az: "{{ env }}-{{ azone }}"
+    env_cf_data_version: "20"
+    env_cf_app_version: "01"
+    build_subnet_id: null
+  vars_files:
+    - "./../../vars/common.yml"
+    - "./../../vault/env/{{ env }}/vault.yml"
+    - "./../../vars/env/{{ env }}/env.yml"
+    - "./../../vars/all_var.yml"
+
+  pre_tasks:
+    - name: "Create ephemeral EC2 instance to run playbook"
+      ec2:
+        instance_type: "{{ cf_app_instance_type }}"
+        group: "BB-SG-{{ env|upper }}-{{ sg_zone|upper }}-ALLZONE" # Change the security group name here
+        image: "{{ ami_app_gold_image }}" # Change the AMI, from which you want to launch the server
+        key_name: "{{ cf_app_key_name }}" # Change the keypair name
+        region: "{{ aws_region }}"
+        tenancy: "dedicated"
+        vpc_subnet_id: "{{ build_subnet_id }}"
+        count: 1
+        termination_protection: no
+        wait: yes
+        wait_timeout: 500
+        user_data: "{{ lookup('file', 'user_data.sh') }}"
+
+        # tags to instance_tags
+        instance_tags:
+          Name: "bb-{{ env }}-{{ azone }}-run-playbook"
+          Stack: "BB-{{ env|upper }}-{{ cf_app_azone|upper }}-{{ cf_app_tag_key_layer|upper }}-ANSIBLE"
+          Business: "{{ cf_tags_business }}"
+          Application: "{{ cf_tags_application }}"
+          Environment: "{{ env|upper }}"
+          Function: "{{ cf_app_tag_key_layer }}-AppServer"
+          Layer:  "{{ cf_app_tag_key_layer|upper }}"
+          ami_name: "BB-{{ env|upper }}-{{ cf_app_tag_key_layer|upper }}"
+          region: "{{ cf_region }}"
+          Managed: "BB-MANAGED-{{ env|upper }}"
+          State: "ami-build"
+      register: app_ec2_output
+
+    - name: "Add new instance to inventory ( {{ app_ec2_output.instances[0]['private_ip'] }} ) "
+      add_host:
+        name: "{{ app_ec2_output.instances[0]['private_ip'] }}"
+        groups: run_servers
+
+    - name: Wait 180 seconds, but only start checking after 60 seconds
+      wait_for_connection:
+        delay: 60
+        timeout: 180
+
+    - name: "Wait for SSH to become available"
+      wait_for:
+        port: 22
+        host: "{{ app_ec2_output.instances[0]['private_ip'] }}"
+        delay: 5
+
+
+- name: Provision EC2 instance
+  hosts: run_servers
+  remote_user: ec2-user
+  gather_facts: no
+  vars:
+    ansible_ssh_pipelining: no
+
+    #roles:
```

Comment:
An opportunity for some formatting, clean up unused/commented code.

Reply:
Going to add back in the security patching. Was commenting that out to save time with testing.